core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitUtils.java

```diff
@@ -460,10 +467,13 @@ static Scan prepareScanForStorage(Scan scan, TableMetadata metadata) {
    *
    * @param conjunctions the conjunctions to convert
    * @param metadata the table metadata of the target table
+   * @param convertIndexedColumns whether to convert conditions on indexed columns
    * @return the converted conjunctions
    */
   private static Set<AndConditionSet> convertConjunctions(
```

Comment:
Modified ConsensusCommitUtils.convertConjunctions():
 - Added convertIndexedColumns parameter to control whether conditions on indexed columns should be converted to include before-column conditions
- For cross-partition scans (ScanAll), indexed column conditions are NOT converted (passed as-is)
- For partition scans and Gets, all conditions including indexed columns ARE converted